### DISCUSSION TITLE
[Question]: Reference object of keys

### DISCUSSION THREAD

**Discussion by cthrax** [ngneat/transloco#467] [Q&A]
Hello, I'm very recently migrating over from ngx-translate and something we used a lot in that library was references to other keys and in some cases, a whole object of keys. It seems like that is not working in transloco, but maybe we need to be doing something differently.

As an example of what I mean:
```
{
   'app': {
       'scope1': {
           'features': {
               'feature1': 'value1',
               'feature2': 'value2'
           }
       },
       'scope2': {
          'features': '{{app.scope1.features}}'
       }
   }
}
```

Thanks in advance!

**Comment by shaharkazaz** [comment]
@cthrax What would you expect in such a case?

**Comment by cthrax** [comment]
@shaharkazaz Sure! I should have mentioned that. So I would expect that translating the key "app.scope2.features.feature1" I get the value "value1". Right now I'm getting the untranslated key.

**Reply by cthrax** [reply]
@shaharkazaz Is my example clear enough for what I'm attempting to accomplish?

**Comment by shaharkazaz** [comment]
@cthrax, your example was clear enough 🙂 
I don't see the reason why reference one scope to the other. If you have common translation keys, put them in the translation's root and use them directly. I don't think it's a good idea to couple scopes; that's my 2 cents about it.

Currently, that's not something I think we will add to the library. 
You can still achieve what you are looking for by extending the transloco transpiler and implementing the logic you need there.

Hope this answers your question.

**Reply by cthrax** [reply]
That's totally fair if it's not a feature for the library. To expand on my use case just a little, in case it's helpful to see how consumers are using the library, here's some more context as to why we ran into this use case.

We structure our translations to follow code organization. We do have a global space up at the root for things like buttons, but it's rare that we have two components in separate parts of the app that share translations... except in this case where one part of our app is an advertising page for customers that is presented to the world and the other is a help item for customers that are logged in. They both live in separate lazy loaded angular modules, but are identical textual pieces for clarity and consistency. Because our keys are ordered mirroring code structure, we look for the keys in two different places, but we want the text to be the same and the translations to be the same. We are accomplishing this now by duplicating each terminal key `app.scope2.features.feature1: "{{app.scope1.features.feature1}}"`, but since every terminal key is going to be the same, we wanted to reference the whole block so there's only one location that those translations need to be managed and it's automatically picked up by the other. 

I think this use case still "couples the scopes" as you say, but is working in the system right now. I imagine it might not work if we create a multi-file scope? Or lazy load some of the translations? Which is probably where you were going. Is that correct?
